src/discussions/posts/data/slices.js

```diff
@@ -1,18 +1,35 @@
 /* eslint-disable no-param-reassign,import/prefer-default-export */
 import { createSlice } from '@reduxjs/toolkit';
-import { RequestStatus } from '../../../data/constants';
+
+import {
+  AllPostsFilter,
+  MyPostsFilter,
+  PostsStatusFilter,
+  RequestStatus,
+  ThreadOrdering,
+} from '../../../data/constants';
+
+function normaliseProfileImage(currentThread, newThread) {
+  newThread.authorAvatars = newThread.users
+    ? newThread.users?.[newThread.author].profile.image
+    : currentThread.authorAvatars;
+  return newThread;
+}
 
 function normaliseThreads(state, rawThreadsData) {
-  const { topicThreadMap: topics, threads } = state;
+  const {
+    topicThreadMap: topics,
+    threads,
+  } = state;
   rawThreadsData.forEach(
     thread => {
-      if (!topics[thread.topic_id]) {
-        topics[thread.topic_id] = [];
+      if (!topics[thread.topicId]) {
+        topics[thread.topicId] = [];
       }
-      if (!topics[thread.topic_id].includes(thread.id)) {
-        topics[thread.topic_id].push(thread.id);
+      if (!topics[thread.topicId].includes(thread.id)) {
+        topics[thread.topicId].push(thread.id);
       }
-      threads[thread.id] = thread;
+      threads[thread.id] = normaliseProfileImage(threads[thread.id], thread);
     },
   );
 }
```

Comment:
I think this will be changed with normalization changes. making it as a todo item